.github/workflows/clp-artifact-build.yaml

```
@@ -164,71 +166,72 @@ jobs:
 
-  manylinux_2_28-x86_64-deps-image:
-    name: "manylinux_2_28-x86_64-deps-image"
-    if: "needs.filter-relevant-changes.outputs.manylinux_2_28_x86_64_image_changed == 'true'"
-    needs: "filter-relevant-changes"
+  deps-image-merge:
+    name: "${{matrix.os}}-deps-image-merge"
+    # Only merge per-arch tags into a multi-arch ":main" manifest on push to main; on PRs the
+    # deps image is consumed as a local tar artifact, so no merge is needed. Each leg is gated
+    # on its own deps-image matrix result, so a failure or skip of one OS doesn't block the other.
+    if: >-
+      github.event_name != 'pull_request'
```

actionlint (1.7.12)

[error] context "matrix" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)